mesonbuild/modules/gnome.py

```diff
@@ -2080,7 +2081,16 @@ def _extract_vapi_packages(self, state: 'ModuleState', packages: T.List[T.Union[
         vapi_args: T.List[str] = []
         remaining_args = []
         for arg in packages:
-            if isinstance(arg, InternalDependency):
+            if isinstance(arg, PkgConfigDependency):
+                vapi_args.append(f'--pkg={arg.name}')
+                vapi_packages.append(arg.name)
+                remaining_args.append(arg.name)
+            elif isinstance(arg, ExternalLibrary):
+                # We can only handle VAPI files here
+                assert arg.language == 'vala'
```

Comment:
Thanks, that definitely makes sense.

> This should raise a suitable exception with an error message such as "erm, generate_vapi can only depend on find_library results if you specifically used the vala language"

Would it make sense that we keep the assert in this helper function while raising the exception at the generate_vapi() level by doing a check there? I'd like to re-use this at some point for the functionality of a new valadoc() API as well (#7177) , so if we were to put function name, I guess it would have to be done like that?